File: ios/maimaid/Models/ScannerNoteCountValidator.swift
nonisolated enum ScannerNoteCountValidator {
    static func isCompatible(maxDxScore: Int?, sheetTotal: Int?) -> Bool {
        guard let maxDxScore, maxDxScore > 0, let sheetTotal else {
            return true
        }

        return maxDxScore / 3 <= sheetTotal
    }
}

File: android/app/src/main/java/org/rhythmeta/maimaid/core/ml/ScannerNoteCountValidator.kt
package org.rhythmeta.maimaid.core.ml

internal object ScannerNoteCountValidator {
    fun isCompatible(maxDxScore: Int?, sheetTotal: Int?): Boolean {
        if (maxDxScore == null || maxDxScore <= 0 || sheetTotal == null) return true
        return maxDxScore / 3 <= sheetTotal
    }
}
